plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -131,6 +131,26 @@ public static Library collectLibraryFiles(IJavaProject javaProject, List<Library
       subMonitor.worked(1);
     }
 
+    // need to get old master library entries first if they exist
```

Comment:
This doesn't seem like the right place: it means there's no way for the user to unselected and remove the libraries. I thought this would go into the `CloudLibrariesPage` subclasses.

Reply:
Correct. That's what #2400 is about. We're not going to implement that here or today. One step at a time.